torch/csrc/lazy/core/tensor_impl.cpp

```diff
@@ -129,7 +146,8 @@ void LTCTensorImpl::shallow_copy_from(
 int64_t LTCTensorImpl::size(int64_t d) const {
   // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
   const_cast<LTCTensorImpl*>(this)->setup_size_properties();
-  return c10::TensorImpl::size(d);
+  //return c10::TensorImpl::size(d);
+  return upper_bounds_.at(d);
```

Comment:
hmm, this effectively change the definition of the size, I think it should be fine for us. Pytorch/xla is not using `LTCTensorImpl` so it won't affect us right us. I expect when we migrate to this class we should be done with the codegen stuff so we should have `lazy::shape`  ready for every op we supported.

Reply:
If possible, I'd advocate for adding a separate method for querying upper bound and then rewriting code to make use for it. But I don't know how much work this is.

Reply:
are there updated design notes?  does upper_bounds_ get precomputed as an int (it's totally separate from the symint stored in sizes_and_strides_)?

Reply:
https://docs.google.com/document/d/18IBeJvccvnEvAfCEqOoPPwdRTEM8CG5i8QGdah0wUNs/edit is sort of half of it but not enough text